socketio/src/packet.rs

```diff
@@ -25,7 +25,7 @@ pub struct Packet {
     pub packet_type: PacketId,
     pub nsp: String,
     pub data: Option<String>,
-    pub id: Option<i32>,
+    pub id: Option<AckId>,
```

Comment:
While I'm generally a fan of wrapper types, it's not worth the breaking API change.

Reply:
the reason I put this in is just so you can't make a mistake when calling `ack` on the client. Since you're not supposed to care what the ack id is I think this is a good change and it's better to make this change sooner rather than later. If you still disagree I can drop it

Reply:
I agree it's a reasonable change, but it would need to be made (and merged) separately.

Reply:
I also agree that this is a reasonable change, but please factor It out in separate commit / PR. This breaks existing users that depend on an `i32` in this place and we cannot just ignore that. That being said, thanks for spotting this appropriately :)